src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -319,13 +319,10 @@ class GoogleExtensions(
           for {
             _ <- assertProjectInTerraOrg(project)
             sa <- IO.fromFuture(IO(googleIamDAO.createServiceAccount(project, petSaName, petSaDisplayName)))
-            // Sleep added to avoid Google race condition where the GoogleDirectoryDAO doesn't yet know about the
-            // newly created service account which leads to slow permission propagation.
-            // See https://broadworkbench.atlassian.net/browse/QA-723 for details
-            // TODO: remove sleep when Google fix exists - see https://broadworkbench.atlassian.net/browse/CA-1005
-            _ <- IO.sleep(2 seconds)(IO.timer(executionContext))
-            r <- IO.fromFuture(IO(withProxyEmail(user.id) { proxyEmail =>
-              googleDirectoryDAO.addMemberToGroup(proxyEmail, sa.email)
+            _ <- IO.fromFuture(IO(withProxyEmail(user.id) { proxyEmail =>
+              // Add group member by uniqueId instead of email to avoid race condition
+              // See: https://broadworkbench.atlassian.net/browse/CA-1005
+              googleDirectoryDAO.addServiceAccountToGroup(proxyEmail, sa)
```

Comment:
I see the corresponding change here https://github.com/broadinstitute/workbench-libs/pull/352/files#diff-09d2ace9afc01f2b2fcded01ac2a7ef0R160 👍

Reply:
Correctomundo!